core/internal/auth/netrc.go

```diff
@@ -104,6 +105,9 @@ func NetrcPath() (string, error) {
 }
 
 func ReadNetrc() ([]netrcLine, error) {
+	netrcMu.Lock()
+	defer netrcMu.Unlock()
```

Comment:
what's this guarding? isn't this only needed for the line that modifies `netrc`?

Reply:
yes. move it there?

Reply:
yes

Reply:
on second thought, we shouldn't need this at all. Just get rid of the global variable since we never read it.